pkg/operator/resource_reconciler.go

```diff
@@ -457,3 +471,14 @@ func ListMatchingNamespaces(selector labels.Selector, nsInf cache.SharedIndexInf
 	}
 	return ns, nil
 }
+
+// isOwnershipWrong returns false if the controller is the "owner" of the prometheus
+// or ownership has not been setted yet
+// Whether it's owner is determined by the value of label 'controllerID'
+// annotation. If the value matches the controllerID then it owns it.
+func (rr *ResourceReconciler) isOwnershipWrong(labels map[string]string) bool {
```

Comment:
(suggestion) let's pass the object metadata
```suggestion
func (rr *ResourceReconciler) isManagedBy(o metav1.Object) bool {
```